test/abstract_syntax_tree.coffee

```diff
@@ -95,1086 +95,1188 @@ test 'Confirm functionality of `deepStrictEqualExpectedProps`', ->
 # the type and properties match. When relevant, also check that values of
 # properties are as expected.
 
-test "AST as expected for Block node", ->
-  deepStrictEqualExpectedProps CoffeeScript.compile('return', ast: yes),
-    type: 'Block'
-    expressions: [
-      type: 'Return'
-    ]
-
-# Can’t test for `Literal` node, as it’s a base class for the `*Literal`
-# classes.
+# test "AST as expected for Block node", ->
+#   deepStrictEqualExpectedProps CoffeeScript.compile('return', ast: yes),
+#     type: 'Block'
+#     expressions: [
+#       type: 'Return'
+#     ]
 
 test "AST as expected for NumberLiteral node", ->
   testExpression '42',
-    type: 'NumberLiteral'
-    value: '42'
+    type: 'NumericLiteral'
+    value: 42
+    extra:
+      rawValue: 42
+      raw: '42'
+    start: 0
+    end: 2
+    range: [0, 2]
+    loc:
+      start:
+        line: 1
+        column: 0
+      end:
+        line: 1
+        column: 2
```

Comment:
@GeoffreyBooth ok you're right it makes sense to separate out location-data testing from other stuff. I do think it makes sense to have AST location data tests for every node type, as we'll see that there are a fair number of places where existing location data wasn't accurate/comprehensive, and eg Prettier and ESLint both get cranky when your location data is off (to be specific, range-based location data seems to be what they both rely on/trust rather than line/column)

Reply:
@GeoffreyBooth extracted AST location data tests into `test/ast_location_data.coffee`, extracted shared helpers into `test/support/helpers.coffee`